src/mots/bmo.py

```diff
@@ -0,0 +1,108 @@
+"""Module that provides helpers to interact with the Bugzilla API."""
+
+# TODO: this should be abstracted so that it can be hot-swappable with any API that can
+# be searched.
+
+from __future__ import annotations
+import requests
+import os
+import logging
+import time
+
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_BASE_URL = "https://bugzilla.mozilla.org/rest"
+USER_AGENT = "mots"  # TODO: improve this and include version.
+
+# TODO set NO_INPUT var to prevent prompt.
+
+
+class BMOClient:
+    """A thin wrapper as a Bugzilla API client."""
+
+    def __init__(self, token: str = None, base_url: str = DEFAULT_BASE_URL):
+        if not token:
+            token = os.getenv("BUGZILLA_API_KEY", "")
+            if not token:
+                token = input("Enter BMO Token: ")
+            if not token:
+                raise Exception()
+        self.headers = {"X-BUGZILLA-API-KEY": token, "User-Agent": USER_AGENT}
+        self.base_url = base_url
+
+    def _get(self, path: str, params=None):
+        logger.debug(f"GET {self.base_url}/{path}")
+        params = params or {}
+        response = requests.get(
+            f"{self.base_url}/{path}", headers=self.headers, params=params
+        )
+        return response
+
+    def get_user(self, email: str, rate_limit_delay: float = 0.3):
+        """Get user data based on provided email address."""
+        time.sleep(rate_limit_delay)
+        fields = ["real_name", "nick", "email", "name", "id"]
+        response = self._get(
+            "user", {"names": email, "include_fields": ",".join(fields)}
+        )
+        if response.status_code != 200:
+            logger.error(f"Error searching {email}: {response.status_code}")
+            return
+
+        # TODO: handle case when there is more than one match.
+        # Since possibly users can sneak in their email address in another field?
+        # Although we should validate the "email" field in BMO response + check for
+        # verification.
+
+        result = response.json()
+        return result["users"][0]
+
+    def get_user_by_id(self, id_: str = "", rate_limit_delay: float = 0.3):
+        """Get user data by BMO ID."""
+        time.sleep(rate_limit_delay)
+        fields = ["real_name", "nick", "name", "id"]
```

Comment:
We probably want to return the same `user` data structure, regardless of which `get_user_by_*` is being called.
Perhaps we should reuse `fields`?

Reply:
This is also not being used any more. Will delete it.